whackaworm.js

```diff
@@ -1,123 +1,88 @@
+	score = 0;
+	previousRandomNum = 0;
+	randomNum = 0;
+	hole = "hole.png";
+	worm = "wormHoleImage.png";
+	holes = [];
 }
 
-function randomize(e)
-{
-	var source = e.target ||event.srcElement;
-	var img = source.src;
-
-	if(img.substring(img.lastIndexOf("/")+1) == worm){
-
-	score+=10;
-	play();
+function cacheImages(){
+    //Cache all the image tags
+	for(i = 1; i <= numOfHoles; i++){
+		holes.push(document.getElementById("hole"+ i.toString()));
 	}
 }
 
-function play()
-{
-	generateNum();
-	imgArr[randomNumber].src = worm;
-	imgArr[previousRandomNum].src = "hole.png";
-	previousRandomNum = randomNumber;
+function addEvent(objName, type, fnName, Cap) {
+	  if(objName.attachEvent){
+		objName.attachEvent("on"+type,fnName);
+	  }
+	  else if(objName.addEventListener){
+		objName.addEventListener(type,fnName,Cap);
+	  }
 }
 
 function endGame(){
-	removeEventHandler(hole1, "click", randomize);
-	removeEventHandler(hole2, "click", randomize);
-	removeEventHandler(hole3, "click", randomize);
-	removeEventHandler(hole4, "click", randomize);
-	removeEventHandler(hole5, "click", randomize);
-	removeEventHandler(hole6, "click", randomize);
-	removeEventHandler(hole7, "click", randomize);
-	removeEventHandler(hole8, "click", randomize);
-	removeEventHandler(hole9, "click", randomize);
+	for(i=0; i < holes.length;i++){
+		removeEventHandler(holes[i], clickAddEventFunction, randomize);
+	}
 	
-	for(var i = 0; i<imgArr.length; i++)
-	{
-		imgArr[i].src = "hole.png";
+	for(var i = 0; i < holes.length; i++){
+		holes[i].src = hole;
 	}
 	alert("Game Over! Final Score: " + score);
 	var btn = document.getElementById("startBtn");
 	btn.style.display = 'block';
 }
 
-function removeEventHandler(elem,eventType,handler) {
- if (elem.removeEventListener) 
-    elem.removeEventListener (eventType,handler,false);
- if (elem.detachEvent)
-    elem.detachEvent ('on'+eventType,handler); 
+function generateNum(){
+	randomNum = Math.floor(Math.random()*10);
+	while(previousRandomNum == randomNum || randomNum == 9){
+		randomNum = Math.floor(Math.random()*10);
+	}
 }
 
-function cacheImages()
-{  
-  
-    //Cache all the image tags
-    hole1 = document.getElementById("hole1");
-    hole2 = document.getElementById("hole2");
-    hole3 = document.getElementById("hole3");
-    hole4 = document.getElementById("hole4");
-    hole5 = document.getElementById("hole5");
-    hole6 = document.getElementById("hole6");
-    hole7 = document.getElementById("hole7");
-    hole8 = document.getElementById("hole8");
-    hole9 = document.getElementById("hole9");
-	worm = "wormHoleImage.png";
-	
-	imgArr = [hole1, hole2, hole3, hole4 ,hole5 ,hole6 ,hole7 ,hole8 ,hole9];
+function play(){
+	generateNum();
+	holes[randomNum].src = worm;
+	holes[previousRandomNum].src = hole;
+	previousRandomNum = randomNum;
+}
+
+function randomize(event){
+	var source = event.target || event.srcElement;
+	var img = source.src;
+
+	if(img.substring(img.lastIndexOf("/")+1) == worm){
+
+	score+=10;
```

Comment:
in this case, in order to change points, I'd have to look into the "randomize" function for a score+= 10. It's not super indicative.